### PR TITLE
Source Iterable: fix invalid chunk length error during discovery

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
*Discovering schema failed/InvalidChunkLength(got length b'', 0 bytes read*
*https://github.com/airbytehq/oncall/issues/3096*

## How
*Disable cache for ListUser stream and set `{"stream": True}` for request*

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-iterable test report (commit [`385aba83f7`](https://github.com/airbytehq/airbyte/pull/31331/commits/385aba83f76e501c90d8ecef2931ad2cec7e3588)) - ✅

⏲️  Total pipeline duration: 04mn35s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-iterable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for source-iterable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-iterable-fix-discovery-invalid-chunk-length/1697119910/385aba83f76e501c90d8ecef2931ad2cec7e3588/source-iterable/0.1.31/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:385aba83f76e501c90d8ecef2931ad2cec7e3588)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-iterable test
```

**Review by bazarnov — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-iterable test report (commit [`b85b7f19ca`](https://github.com/airbytehq/airbyte/pull/31331/commits/b85b7f19ca065fffae9e5689ffe14b52ce828ccc)) - ✅

⏲️  Total pipeline duration: 03mn12s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build source-iterable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for source-iterable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/ryermilov_source-iterable-fix-discovery-invalid-chunk-length/1697121375/b85b7f19ca065fffae9e5689ffe14b52ce828ccc/source-iterable/0.1.31/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b85b7f19ca065fffae9e5689ffe14b52ce828ccc)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-iterable test
```

File: airbyte-integrations/connectors/source-iterable/source_iterable/streams.py [hunk 1]
```diff
@@ -354,11 +354,29 @@ class ListUsers(IterableStream):
     data_field = "getUsers"
     name = "list_users"
     # enable caching, because this stream used by other ones
-    use_cache = True
+    use_cache = False
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-iterable/source_iterable/streams.py:357]: shouldn't we leave it as cacheable stream?

Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-iterable/source_iterable/streams.py:357]: In this case {stream: True} has no any effect

Comment (artem1205) [reply airbyte-integrations/connectors/source-iterable/source_iterable/streams.py:357]: but don't we use List Users stream as parent for Events ?

**Review by lazebnyi — APPROVED** [review]
Lgtm after Artem requested updates

**Pull request closed without merging**